sycl/source/handler.cpp

```diff
@@ -383,11 +383,11 @@ event handler::finalize() {
     std::shared_ptr<ext::oneapi::experimental::detail::graph_impl> GraphImpl;
     if (MGraph) {
       GraphImpl = MGraph;
+      CommandGroup.reset(
+          new detail::CG(detail::CG::None, std::move(CGData), MCodeLoc));
+      MGraphNodeCG = std::move(CommandGroup);
     } else if (auto QueueGraph = MQueue->getCommandGraph(); QueueGraph) {
       GraphImpl = QueueGraph;
-    }
-
-    if (GraphImpl) {
       auto EventImpl = std::make_shared<detail::event_impl>();
```

Comment:
I think you can remove `GraphImpl` here and just use `QueueGraph` in that branch of the `if`, since `MGraph` is only used for the conditional anyway.

Reply:
Applied.